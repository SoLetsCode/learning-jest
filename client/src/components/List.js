import React from "react";

export default function List() {
  const listArray = [];

  for (let i = 0; i < 6; i++) {
    listArray.push(<li key={i}>{i}</li>);
  }
  return (
    <div>
      <ul>{listArray}</ul>
    </div>
  );
}
