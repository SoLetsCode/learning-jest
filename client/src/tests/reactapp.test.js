import React from "react";
import List from "../components/List";
import App from "../App";
import renderer from "react-test-renderer";

test("list compiles correctly", () => {
  const tree = renderer.create(<List></List>).toJSON();

  expect(tree).toMatchSnapshot();
});

test("snapshot test of app", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
