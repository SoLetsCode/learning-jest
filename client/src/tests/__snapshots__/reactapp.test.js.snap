// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list compiles correctly 1`] = `
<div>
  <ul>
    <li>
      0
    </li>
    <li>
      1
    </li>
    <li>
      2
    </li>
    <li>
      3
    </li>
    <li>
      4
    </li>
    <li>
      5
    </li>
  </ul>
</div>
`;

exports[`snapshot test of app 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
    <div>
      <ul>
        <li>
          0
        </li>
        <li>
          1
        </li>
        <li>
          2
        </li>
        <li>
          3
        </li>
        <li>
          4
        </li>
        <li>
          5
        </li>
      </ul>
    </div>
  </header>
</div>
`;
